packages/core/src/components/Autocomplete/index.tsx

```diff
@@ -536,6 +536,10 @@ export default class Autocomplete<T extends Item = Item> extends React.Component
           loading: false,
         };
 
+        if (input !== this.state.value) {
+          delete nextState['items'];
+        }
```

Comment:
Were you able to functionally test this? `setState` is async and batches requests, so it's possible to be out of sync, you could shadow the value of `this.state.value` to a class property to be 100% sure that `input` is current. 

I think worst case is `this.state.value` falls out of sync is you'd have the same behavior you're fixing, and react would have to be in a bad state to get that far down on update state.